theano/scan_module/scan_utils.py

```diff
@@ -54,28 +55,14 @@ def safe_new(x, tag='', dtype=None):
             return nwx
         else:
             return x.clone()
-    # Note, as_tensor_variable will convert the Scalar into a
-    # TensorScalar that will require a ScalarFromTensor op,
-    # making the pushout optimization fail
-    elif isinstance(x, scalar.ScalarVariable):
-        if dtype:
-            nw_x = scalar.get_scalar_type(dtype=dtype)()
-        else:
-            nw_x = x.type()
-        nw_x.name = nw_name
-        return nw_x
-    else:
-        try:
-            x = tensor.as_tensor_variable(x)
-        except TypeError:
-            # This could happen for example for random states, and I really
-            # want to avoid the convoluted logic that checks for cuda
-            # ndarrays
-            pass
```

Comment:
Because the as_tensor_variable() is a workaround for one place where we call safe_new() with an ndarray and the ScalarVariable case is a workaround for as_tensor_variable() on scalars.

Fixing the one call site which actually uses ndarray instead is cleaner and safer.